lib/facter/facts_utils/virtual_detector.rb

```diff
@@ -58,6 +59,13 @@ def check_xen
       Facter::Resolvers::Xen.resolve(:vm)
     end
 
+    def check_freebsd
+      return unless Object.const_defined?('Facter::Resolvers::Freebsd::Virtual')
```

Comment:
I had to use that trick so that the test suite pass on Linux (CI suite on GitHub).  The file already use `Facter::Resolvers::Linux::DmiBios` which does not cause trouble running the test suite on FreeBSD, so I guess something magic can be done somewhere to improve this situation :wink:

Reply:
You can try something like: `return unless Facter::Resolvers::Uname.resolve(:kernelname) == 'FreeBSD'` (not sure if the kernel name is exactly that string)

Reply:
I guess it will make things worst if at some point support is added for an OS which is a FreeBSD derivative, e.g. HardenestBSD.  In such a case, this system would see and could use the FreeBSD classes, like for example the FreeBSD classes could see and use the Solaris ones when FreeBSD was (incorrectly) a Solaris derivative.

Maybe if there is no better trick for now, this can stay as is?

Reply:
same for Dragonfly, but i don't know how many people use that on servers

we need some kind of "family"?

Reply:
can you not a HardenedBSD and show me what (the different versions of) `facter os` look like?

Reply:
We decide which hierarchy to load based on RbConfig::CONFIG['host_os'] (in case of FreeBSD the result is /freebsd/ https://github.com/puppetlabs/facter/blob/21c0f1a94b79c63f520f42317b6499e15b862084/lib/facter/framework/detector/os_detector.rb#L19) maybe you could use that, if there are scenarios when that is wrong also, I'm ok to leave things as they are.

Reply:
@igalic, I do not have a HardenedBSD at hand, it's still on my list to things to do :yum: I similarly do not have any DragonflyBSD at hand, but I guess both systems would be detected as "bsd" and adjusting `RbConfig::CONFIG['host_os']` as suggested by @oanatmaria might be useful to improve what facter returns?

Adding support for these os is low priority for me now, so I prefer we keep it that way for now and maybe improve this if required when support of these derivative is added to facter.

If it's okay for you, we can resolve this conversation and the PR can be merged :tada: